module/VuFind/tests/integration-tests/src/VuFindTest/Mink/OnlinePaymentTest.php

```diff
@@ -0,0 +1,763 @@
+     *
+     * @param bool   $multibackend    Use MultiBackend driver?
+     * @param ?array $paymentSettings Payment settings, or null to disable
+     *
+     * @return array
+     */
+    protected function getConfigs(bool $multibackend, ?array $paymentSettings = null): array
+    {
+        $configs = [
+            'config' => $this->getConfigIniOverrides($multibackend),
+        ];
+        $paymentOverrides = null !== $paymentSettings
+            ? $this->getDemoIniOverridesForPayment($paymentSettings)
+            : [];
+        if ($multibackend) {
+            $configs['MultiBackend'] = [
+                'Drivers' => [
+                    'pay' => 'Demo',
+                    'nopay' => 'Demo',
+                ],
+                'Login' => [
+                    'default_driver' => 'pay',
+                    'drivers' => [
+                        'pay',
+                        'nopay',
+                    ],
+                ],
+            ];
+            $configs['Demo:pay'] = $this->getDemoIniOverrides() + $paymentOverrides;
+            $configs['Demo:nopay'] = $this->getDemoIniOverrides();
+        } else {
+            $configs['Demo'] = $this->getDemoIniOverrides() + $paymentOverrides;
+        }
+        return $configs;
+    }
+
+    /**
+     * Get config.ini override settings for testing payment functions.
+     *
+     * @param bool $multibackend Use MultiBackend driver?
+     *
+     * @return array
+     */
+    protected function getConfigIniOverrides(bool $multibackend): array
+    {
+        $config = [
+            'Catalog' => [
+                'driver' => $multibackend ? 'MultiBackend' : 'Demo',
+                'holds_mode' => 'driver',   // needed to display login link
+            ],
+            'Mail' => [
+                'testOnly' => true,
+                'message_log' => $this->getEmailLogPath(),
+                'message_log_format' => $this->getEmailLogFormat(),
+            ],
+        ];
+        return $config;
+    }
+
+    /**
+     * Get Demo.ini override settings for enabling payment.
+     *
+     * @param array $additional Additional settings
+     *
+     * @return array
+     */
+    protected function getDemoIniOverridesForPayment(array $additional = []): array
+    {
+        return [
+            'OnlinePayment' => array_merge(
+                [
+                    'enabled' => true,
+                    'currency' => 'USD',
+                    'selectFines' => true,
+                    'productCodeMappings' => 'Overdue=demo_003:Long Overdue=demo_004',
+                    'handler' => 'Test',
+                    'url' => $this->getVuFindUrl('/devtools/payment'),
+                ],
+                $additional
+            ),
+        ];
+    }
+
+    /**
+     * Get fine JSON for Demo.ini.
+     *
+     * @param string $bibId Bibliographic record ID to create fake item info for.
+     *
+     * @return array
+     */
+    protected function getFakeFines(string $bibId)
```

Comment:
Missing return type.

```suggestion
    protected function getFakeFines(string $bibId): array
```